source/Conversation.h

```diff
@@ -32,6 +32,17 @@ class Sprite;
 // killed. A conversation can also branch based on various condition flags that
 // are set for the player, and can also modify those flags.
 class Conversation {
+	class Data {
+	public:
+		// The text to display:
+		std::string text; 
+		// The next node to visit:
+		int next;
+		// Conditions for displaying the text:
+		ConditionSet conditions; 
+		explicit Data(std::string text, int next, ConditionSet conditions)
+			: text(text), next(next), conditions(conditions) {}
```

Comment:
If all calls to the explicit `Conversation::Data` constructor are passing in an empty, default-constructed `ConditionSet`, then you don't need that parameter in the explicit constructor.


```suggestion
		explicit Data(std::string text, int next)
			: text(std::move(text)), next(next) {}
```
It's also common for our codebase to place constructors before fields, even when the fields are public.

Reply:
> If all calls to the explicit `Conversation::Data` constructor are passing in an empty, default-constructed `ConditionSet`, then you don't need that parameter in the explicit constructor.

This change makes me feel weird, since I really wanted this to just be a 3-tuple, but... you raise a very good point.

> It's also common for our codebase to place constructors before fields, even when the fields are public.

I'll move that before I do anything else (so I don't forget).